SpatialGDK/Source/SpatialGDKServices/Private/SpatialPackageManager.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Async/Async.h"
+#include "SpatialPackageManager.h"
+#include "SpatialGDKServicesConstants.h"
+
+
+DEFINE_LOG_CATEGORY(LogSpatialPackageManager);
+
+
+FSpatialPackageManager::FSpatialPackageManager() {}
+
+
+void FSpatialPackageManager::TryFetchRuntimeBinary(FString RuntimeVersion)
+{
+
+	FString RuntimePath = FString::Printf(TEXT("%s/runtime/%s"), *SpatialGDKServicesConstants::GDKProgramPath, *RuntimeVersion);
+
+	// Check if the binary already exists for a given version
+	if (FPaths::FileExists(FString::Printf(TEXT("%s/%s"),*RuntimePath, *SpatialGDKServicesConstants::RuntimeExe)))
+	{
+		UE_LOG(LogSpatialPackageManager, Verbose, TEXT("Runtime binary already exist."));
+		return;
+	}
+
+	// If it does not exist then fetch the binary using `spatial worker package retrieve`
+	UE_LOG(LogSpatialPackageManager, Log, TEXT("Trying to fetch runtime version %s"), *RuntimeVersion);
+
+	// Download and unzip the executable to "UnrealGDK\SpatialGDK\Binaries\ThirdParty\Improbable\Programs\Runtime\*version*"
+	FString Params =
+		FString::Printf(TEXT("package retrieve runtime %s %s %s --unzip"), *SpatialGDKServicesConstants::PlatformVersion, *RuntimeVersion, *RuntimePath);
+	FSpatialPackageManager FSpatialPackageManager = {};
+	FSpatialPackageManager.StartProcess(Params, "Runtime Fetching");
+}
+
+void FSpatialPackageManager::TryFetchInspectorBinary(FString InspectorVersion)
+{
+
+	FString InspectorPath = FString::Printf(TEXT("%s/inspector/%s"), *SpatialGDKServicesConstants::GDKProgramPath, *InspectorVersion);
+
+	// Check if the binary already exists
+	if (FPaths::FileExists(FString::Printf(TEXT("%s/%s"), *InspectorPath, *SpatialGDKServicesConstants::InspectorExe)))
+	{
+		UE_LOG(LogSpatialPackageManager, Verbose, TEXT("Inspector binaries already exist."));
+		return;
+	}
+
+	// If it does not exist then fetch the binary using `spatial worker package get`
+	// Download the package to // UnrealGDK\SpatialGDK\Binaries\ThirdParty\Improbable\Programs
+	FString Params = FString::Printf(TEXT("package get inspector %s %s %s/%s"), *SpatialGDKServicesConstants::PlatformVersion, *InspectorVersion,
+									 *InspectorPath, *SpatialGDKServicesConstants::InspectorExe);
+	FSpatialPackageManager FSpatialPackageManager = {};
```

Comment:
```suggestion
	FSpatialPackageManager FSpatialPackageManager;
```

does that work as well? not completely sure, but as it is an empty struct I would assume that it would work?

Reply:
Aren't both going to default construct you and object?